### PR TITLE
Release 2.0.0-rc3

### PR DESCRIPTION
**Pull request by erimatnor**
This release candidate contains bugfixes since the previous release
candidate, as well as additional minor features including support for
"user-mapping" authentication between access/data nodes and an
experimental API for refreshing continuous aggregates on individual
chunks.

**Minor Features**
* #2627 Add optional user mappings support
* #2635 Add API to refresh continuous aggregate on chunk

**Bugfixes**
* #2560 Fix SCHEMA DROP CASCADE with continuous aggregates
* #2593 Set explicitly all lock parameters in alter_job
* #2604 Fix chunk creation on hypertables with foreign key constraints
* #2610 Support analyze of internal compression table
* #2612 Optimize internal cagg_watermark function
* #2613 Refresh correct partial during refresh on drop
* #2617 Fix validation of available extensions on data node
* #2619 Fix segfault in decompress_chunk for chunks with dropped columns
* #2620 Fix DROP CASCADE for continuous aggregate
* #2625 Fix subquery errors when using AsyncAppend
* #2626 Fix incorrect total_table_pages setting for compressed scan
* #2628 Stop recursion in cache invalidation

### CODE REVIEW

**Review by mkindahl — APPROVED** [review]

**Comment by mfreed** [comment]
Looks good to me.   

We might want to add an additional sentence around the "minor features",
but I'm okay either way.  Feel free to tweak language below.

```
This release candidate contains bugfixes since the previous release
candidate, as well as additional minor features including support
for "user-mapping" authentication between access and data nodes.
```

I couldn't remember if #2635 was meant to be internal/experimental or public at this point.  If public, we might say:

```
This release candidate contains bugfixes since the previous release
candidate, as well as additional minor features including support
for "user-mapping" authentication between access/data nodes and
an API for refreshing continuous aggregates on individual chunks.
```

**Review by mfreed — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,37 @@
...
+* #2627 Add optional user mappings support
+* #2635 Add API to refresh continuous aggregate on chunk
+
+**Bugfixes**
+* #2593 Set explicitly all lock parameters in alter_job
+* #2604 Fix chunk creation on hypertables with foreign key constraints 
+* #2612 Optimize internal cagg_watermark function
+* #2613 Refresh correct partial during refresh on drop
+* #2617 Fix validation of available extensions on data node
+* #2619 Fix segfault in decompress_chunk for chunks with dropped columns
```
Comment (gayyappan) [inline CHANGELOG.md:25]: Bugfix 2631 is the port of 2619 to 1.7?

Comment (erimatnor) [reply CHANGELOG.md:25]: Yes, good catch.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,37 @@
...
+
+**Bugfixes**
+* #2593 Set explicitly all lock parameters in alter_job
+* #2604 Fix chunk creation on hypertables with foreign key constraints 
+* #2610 Support analyze of internal compression table
+* #2612 Optimize internal cagg_watermark function
+* #2613 Refresh correct partial during refresh on drop
+* #2617 Fix validation of available extensions on data node
+* #2619 Fix segfault in decompress_chunk for chunks with dropped columns 
+* #2620 Fix DROP CASCADE for continuous aggregate
```
Comment (pmwkaa) [inline CHANGELOG.md:27]: Also need this one I believe: https://github.com/timescale/timescaledb/pull/2560  Fix SCHEMA DROP CASCADE with continuous aggregates

**Review by pmwkaa — APPROVED** [review]

**Pull request merged**